Source/Core/VideoBackends/D3D/VertexManager.cpp

```diff
@@ -194,6 +201,9 @@ void VertexManager::ResetBuffer(u32 stride)
 {
 	s_pCurBufferPointer = s_pBaseBufferPointer;
 	IndexGenerator::Start(GetIndexBuffer());
+
+	if (bpmem.genMode.zfreeze)
+		PixelShaderManager::SetZSlope(ZSlope.dfdx, ZSlope.dfdy, ZSlope.f0);
```

Comment:
or if the projection matrix did change, or the transformation one, or the next configuration flag we'll miss. I think we can check it for the dirty flag, but it will be quite hard to skip some of those transformation passes.